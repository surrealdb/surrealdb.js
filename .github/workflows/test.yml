name: test

on:
  push:
  pull_request:

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Setup node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
      - name: Setup deno
        uses: denoland/setup-deno@v1
        with:
          deno-version: v1.x
      - name: Setup bun
        uses: antongolub/action-setup-bun@v1
        with:
          bun-version: 0.1.13

      - uses: actions/checkout@v3

      - name: Build NPM package
        run: deno task build

      - name: Start SurrealDB
        run: sh ./test/e2e/scripts/docker-restart.sh

      - name: Run Deno
        run: deno test -A --trace-ops ./test/e2e/deno.js
      - name: Run Node
        run: node ./test/e2e/node.js
      - name: Run Bun
        run: bun run ./test/e2e/bun.js

      - name: Stop SurrealDB
        run:  sh ./test/e2e/scripts/docker-stop.sh
