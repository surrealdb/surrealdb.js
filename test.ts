import { RecordId } from "./src/library/data/recordid.ts";
import { Duration } from "./src/library/data/duration.ts";
import { UUID } from "./src/library/data/uuid.ts";
import { Decimal } from "./src/library/data/decimal.ts";
import { Surreal } from "./src/index.ts";
import { encode } from "./src/library/data/cbor.ts";
import { GeometryCollection, GeometryLine, GeometryPoint } from "./src/library/data/geometry.ts";
// import { recordId } from "./src/library/orm/types.ts";

const surreal = new Surreal();
await surreal.connect('ws://127.0.0.1:8000/rpc');
await surreal.use({
	namespace: 'test',
	database: 'test',
});

// console.log(
// 	await surreal.query<[unknown[]]>(
// 		"[$d, $r, $c, $u, $dur, $none, NONE, $a, <string> $a, $nan, $geo]",
// 		{
// 			d: new Date(),
// 			r: new RecordId('person', 'tobie'),
// 			c: new RecordId('recording', [
// 				new Date(),
// 				'London',
// 				new Uuid(),
// 				{
// 					temparature: new Decimal("10.00003")
// 				}
// 			]),
// 			u: new Uuid(),
// 			dur: new Duration('1d20m'),
// 			none: undefined,
// 			a: 9223372036854775807n,
// 			nan: NaN,
// 			geo: new GeometryCollection([
// 				new GeometryPoint([1, 2]),
// 				new GeometryLine([ new GeometryPoint([1, 2]), new GeometryPoint([3, 4]) ])
// 			])
// 		}
// 	)
// )

const uuid = new Duration("2w6d100ms");
console.log(
	456,
	await surreal.query<[UUID, string]>(
		"$uuid; <string> $uuid",
		{
			uuid
		}
	).then(([u, a]) => [u.toString(), a])
)

// function bufferToHex (buffer: ArrayBuffer) {
//     return [...new Uint8Array (buffer)]
//         .map (b => b.toString (16).padStart (2, "0"))
//         .join ("");
// }

// console.log(bufferToHex(encode(new RecordId('recording', [
// 	new Date(),
// 	'London',
// 	new Uuid(),
// 	{
// 		temparature: new Decimal("10.00003")
// 	}
// ]))))

// const rid = recordId('test');
// console.log(rid.parse(new RecordId('tesst', 123) ))

// console.log()
