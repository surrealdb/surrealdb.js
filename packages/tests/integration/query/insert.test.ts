import { describe, test, expect } from "bun:test";
import { RecordId } from "surrealdb";
import { setupServer, type Person } from "../__helpers__";

const { createSurreal } = await setupServer();

describe("insert()", async () => {
	const surreal = await createSurreal();

	test("single", async () => {
		const [single] = await surreal.insert<Person>({
			id: new RecordId("person", 1),
			firstname: "John",
			lastname: "Doe",
		});

		expect(single).toStrictEqual({
			id: new RecordId("person", 1),
			firstname: "John",
			lastname: "Doe",
		});
	});

	test("multiple", async () => {
		const multiple = await surreal.insert<Person>([
			{
				id: new RecordId("person", 3),
				firstname: "John",
				lastname: "Doe",
			},
			{
				id: new RecordId("person", 4),
				firstname: "Mary",
				lastname: "Doe",
			},
		]);

		expect(multiple).toStrictEqual([
			{
				id: new RecordId("person", 3),
				firstname: "John",
				lastname: "Doe",
			},
			{
				id: new RecordId("person", 4),
				firstname: "Mary",
				lastname: "Doe",
			},
		]);
	});
});
