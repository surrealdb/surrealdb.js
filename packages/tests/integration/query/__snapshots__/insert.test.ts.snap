// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`insert() compile 1`] = `"INSERT $bind__1"`;

exports[`insert() compile 2`] = `
{
  "bind__1": [
    {
      "firstname": "John",
      "id": RecordId {},
      "lastname": "Doe",
    },
  ],
}
`;
